tools/orca/src/main.rs

```diff
@@ -205,74 +206,88 @@ async fn main() -> ExitCode {
+                        // but not be consumed each loop iteration.
+                        result = &mut run_execute => {
+                            match result {
+                                Ok(_) => {
+                                    return ExitCode::SUCCESS;
+                                }
+                                Err(_err) => {
+                                    return ExitCode::FAILURE;
+                                }
+                            };
+                        }
+                        // Signal handling.
+                        Ok(()) = tokio::signal::ctrl_c() => {
+                            info!("Stopping Task ...");
+                            let _ = control_tx.send(run::Signal::Stop);
+                        }
+                        Some(()) = async move {
+                            let sigterm = tokio::signal::unix::SignalKind::terminate();
+                            #[allow(clippy::unwrap_used)]
+                            tokio::signal::unix::signal(sigterm).unwrap().recv().await
+                        } => {
+                            // Kill it with fire I guess.
+                            return ExitCode::FAILURE;
+                        }
+                        Some(()) = async move {
+                            let sigterm = tokio::signal::unix::SignalKind::alarm();
+                            #[allow(clippy::unwrap_used)]
+                            tokio::signal::unix::signal(sigterm).unwrap().recv().await
+                        } => {
+                            // Ignore
+                        }
+                        Some(()) = async move {
+                            let sigterm = tokio::signal::unix::SignalKind::hangup();
+                            #[allow(clippy::unwrap_used)]
+                            tokio::signal::unix::signal(sigterm).unwrap().recv().await
+                        } => {
+                            // Ignore
+                        }
+                        Some(()) = async move {
+                            let sigterm = tokio::signal::unix::SignalKind::user_defined1();
+                            #[allow(clippy::unwrap_used)]
+                            tokio::signal::unix::signal(sigterm).unwrap().recv().await
+                        } => {
+                            // Ignore
+                        }
+                        Some(()) = async move {
+                            let sigterm = tokio::signal::unix::SignalKind::user_defined2();
+                            #[allow(clippy::unwrap_used)]
+                            tokio::signal::unix::signal(sigterm).unwrap().recv().await
+                        } => {
+                            // Ignore
+                        }
                     }
                 }
-            }
+            })
         }
-    };
+    }
+}
+
+fn build_tokio_runtime(threads: Option<usize>) -> Runtime {
```

Comment:
doc comments to describe how this works with threads whene None means to use the default multi thread scheduler.